test/tstInterpDetailsPolyBasis.cpp

```diff
@@ -0,0 +1,74 @@
+/****************************************************************************
+ * Copyright (c) 2023 by the ArborX authors                                 *
+ * All rights reserved.                                                     *
+ *                                                                          *
+ * This file is part of the ArborX library. ArborX is                       *
+ * distributed under a BSD 3-clause license. For the licensing terms see    *
+ * the LICENSE file in the top-level directory.                             *
+ *                                                                          *
+ * SPDX-License-Identifier: BSD-3-Clause                                    *
+ ****************************************************************************/
+
+#include "ArborX_EnableViewComparison.hpp"
+#include <interpolation/details/ArborX_InterpDetailsPolynomialBasis.hpp>
+
+#include <boost/test/unit_test.hpp>
+
+BOOST_AUTO_TEST_CASE(polynomial_basis_slice_lengths)
+{
+  using view = Kokkos::View<std::size_t **, Kokkos::HostSpace>;
+
+  auto [arr0] =
+      ArborX::Interpolation::Details::polynomialBasisSliceLengths<5, 3>();
+  std::size_t ref0[3][5] = {
+      {1, 1, 1, 1, 1}, {1, 2, 3, 4, 5}, {1, 3, 6, 10, 15}};
+  view arr0_view(&arr0[0][0], 3, 5);
+  view ref0_view(&ref0[0][0], 3, 5);
+  ARBORX_MDVIEW_TEST(arr0_view, ref0_view);
+
+  auto [arr1] =
+      ArborX::Interpolation::Details::polynomialBasisSliceLengths<2, 3>();
+  std::size_t ref1[3][2] = {{1, 1}, {1, 2}, {1, 3}};
+  view arr1_view(&arr1[0][0], 3, 2);
+  view ref1_view(&ref1[0][0], 3, 2);
+  ARBORX_MDVIEW_TEST(arr1_view, ref1_view);
+}
+
+BOOST_AUTO_TEST_CASE(polynomial_basis_size)
+{
+  auto sz0 = ArborX::Interpolation::Details::polynomialBasisSize<5, 3>();
+  BOOST_TEST(sz0 == 56);
```

Comment:
Probably better to do it without defining a variable
```suggestion
  BOOST_TEST(ArborX::Interpolation::Details::polynomialBasisSize<5, 3>(); == 56);
```
If the test fails, instead of cryptic `sz0 == 56` failed, you would immediately see the error.